TRACY_MANUAL_LIFETIME requires enabled TRACY_DELAYED_INIT

Hello!

I have tried to compile TracyClient with enabled TRACY_MANUAL_LIFETIME, but I got the following error:
```gcc
In file included from PATH/tracy/example/build_MAN/_deps/tracy-src/public/TracyClient.cpp:23:
PATH/tracy/example/build_MAN/_deps/tracy-src/public/client/TracyProfiler.cpp: In function 'void ___tracy_startup_profiler()':
PATH/tracy/example/build_MAN/_deps/tracy-src/public/client/TracyProfiler.cpp:4913:12: error: 'StartupProfiler' is not a member of 'tracy'
 4913 |     tracy::StartupProfiler();
      |            ^~~~~~~~~~~~~~~
PATH/tracy/example/build_MAN/_deps/tracy-src/public/client/TracyProfiler.cpp: In function 'void ___tracy_shutdown_profiler()':
PATH/tracy/example/build_MAN/_deps/tracy-src/public/client/TracyProfiler.cpp:4918:12: error: 'ShutdownProfiler' is not a member of 'tracy'
 4918 |     tracy::ShutdownProfiler();
      |            ^~~~~~~~~~~~~~~~
PATH /tracy/example/build_MAN/_deps/tracy-src/public/client/TracyProfiler.cpp: In function 'int ___tracy_profiler_started()':
PATH/tracy/example/build_MAN/_deps/tracy-src/public/client/TracyProfiler.cpp:4923:19: error: 's_isProfilerStarted' is not a member of 'tracy'
 4923 |     return tracy::s_isProfilerStarted.load( std::memory_order_seq_cst );
      |                   ^~~~~~~~~~~~~~~~~~~
```

Enabling TRACY_DELAYED_INIT fixes this issue.

It happens since definition of those functions in only in scope of ifdef TRACY_DELAYED_INIT.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -106,6 +106,11 @@ mark_as_advanced(TRACY_VERBOSE)
 set_option(TRACY_DEMANGLE "[advanced] Don't use default demangling function - You'll need to provide your own" OFF)
 mark_as_advanced(TRACY_DEMANGLE)
 
+# handle incompatible combinations
+if(TRACY_MANUAL_LIFETIME AND NOT TRACY_DELAYED_INIT)
+    message(FATAL_ERROR "TRACY_MANUAL_LIFETIME can not be activated with disabled TRACY_DELAYED_INIT")
+endif()
+
 if(NOT TRACY_STATIC)
     target_compile_definitions(TracyClient PRIVATE TRACY_EXPORTS)
     target_compile_definitions(TracyClient PUBLIC TRACY_IMPORTS)

--- a/public/client/TracyProfiler.cpp
+++ b/public/client/TracyProfiler.cpp
@@ -81,6 +81,10 @@
 #include "TracySysTrace.hpp"
 #include "../tracy/TracyC.h"
 
+#if defined TRACY_MANUAL_LIFETIME && !defined(TRACY_DELAYED_INIT)
+#  error "TRACY_MANUAL_LIFETIME requires enabled TRACY_DELAYED_INIT"
+#endif
+
 #ifdef TRACY_PORT
 #  ifndef TRACY_DATA_PORT
 #    define TRACY_DATA_PORT TRACY_PORT
@@ -4907,7 +4911,7 @@ TRACY_API void ___tracy_fiber_enter( const char* fiber ){ tracy::Profiler::Enter
 TRACY_API void ___tracy_fiber_leave( void ){ tracy::Profiler::LeaveFiber(); }
 #endif
 
-#  ifdef TRACY_MANUAL_LIFETIME
+#  if defined TRACY_MANUAL_LIFETIME && defined TRACY_DELAYED_INIT
 TRACY_API void ___tracy_startup_profiler( void )
 {
     tracy::StartupProfiler();

